### ISSUE TITLE
Implement utility class for removing gold outline during focus

### ISSUE THREAD

**Issue by ncksllvn** [department-of-veterans-affairs/va.gov-team#457]
Throughout vets-website, we are often seeing `outline: none` used to remove a gold outline from elements that have had browser focus applied programmatically, usually via the [focusElement](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/utilities/ui/index.js#L23) helper. The gold outline is applied to any elements that are in the focused state and have a `tabindex`. The `focusElement` helper adds this `tabindex` to that element so that it is focusable via JS. [See this article for more info](https://bitsofco.de/how-and-when-to-use-the-tabindex-attribute/).

The reason we're removing the gold outline from many elements is because it doesn't look visually correct when the focused element is just content, and not a form input or a link.

For example, on the upcoming Disability Calculator, new content is rendered after the form is submitted and focus is applied to that content so a screenreader will read this content automatically. A gold outline is then applied automatically to this content, which is not ideal -

![image.png](https://images.zenhubusercontent.com/59cb9de1b0222d5de47953d8/7f94d90a-6d1b-43d6-be31-e5c7c8e38a17)

So, we have to [add some CSS to remove this style](https://github.com/department-of-veterans-affairs/vets-website/pull/10345/files#diff-c0522ef87ad6d5e0b69411aba3cf4380R25).

----
### Possible solution

To fix this, it might be worth adding a utility class in the design system that removes the outline during the focus state. Then, we can just add that class to any elements that will have focus programmatically applied but should not have a gold outline during the focus state. This utility class could be something like `vads-u-no-focus-outline`.

**Comment by ncksllvn** [comment]
@jaredcunha-usds Curious on your thoughts on the solution proposed here for a design system utility class that remove the `outline` property

**Comment by jaredcunha-usds** [comment]
Hmmm… so, what I'm thinking is that outline is a property that can be applied without focus. It's like adding a border without impacting the box sizing, so it's not always related to accessibility. To create an outline utility properly, we'd have to have to set base versions of the outline, allowing for variants in width, then create a `:focus` state of the utility for all of that.

I wonder if it's just better to have a some tokens for this one until we can determine a larger need?

**Comment by humancompanion-usds** [comment]
@rsmithadhoc - Can you review this and give me your recommendation? Thanks.

**Comment by rsmithadhoc** [comment]
My recommendation is to not add a helper class that removes an outline. Seeing `outline: none;` or `outline: 0;` is typically a red flag for accessibility. There might be some rare cases where it is necessary, but I don't believe this case is one of them.

The element should not be focusable. Typically, only interactive elements should be focusable. From [WebAim](https://webaim.org/techniques/keyboard/#problems)

> Links, buttons, and form controls are natively accessible to keyboard users, so should be used for interactivity whenever possible. Page elements that are not interactive to mouse or touch users should not be made keyboard focusable (such as by using tabindex). Making non-interactive elements keyboard navigable will cause confusion.

To solve this issue of new content being rendered on the page after the form is submitted, [ARIA Live Regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) should be used.

**Comment by humancompanion-usds** [comment]
@rsmithadhoc - Awesome, thanks! Where would you suggest we put that a11y guidance? We only have a11y guidance under components and some patterns. We don't talk about focus globally. Should we add a page for that or does this belong under a set of components (like link and button)?

**Comment by rsmithadhoc** [comment]
@humancompanion-usds Sorry, I missed this comment when originally posted.

I think the guidance should be on a global focus management page. I'm not sure on the best spot for this, doesn't quite fit under our current sections. It could go under **Content**, it's not technically content, but it does fit in with some other a11y-centric pages there like alt text.

I'll close this issue out as not doing, but created [another issue](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3164) to create the documentation page.

**Comment by rsmithadhoc** [comment]
I'm closing this issue. I don't believe it is justified to disable the focus indicator. Keyboard and assistive tech users should be able to know where their focus is. Even when focus is manually set to a heading, I believe it should be shown.
